### PR TITLE
fix: Memoize virtualized renderer

### PR DESCRIPTION
**Pull request by ling1726**
Memoizes the virtualized renderer to avoid react-window mounting continously. Also updates the example to memoize the row renderer.

### CODE REVIEW

File: packages/react-components/react-table/stories/DataGrid/Virtualization.stories.tsx [hunk 1]
```diff
@@ -158,6 +182,15 @@ export const Virtualization = () => {
   const { targetDocument } = useFluent();
   const scrollbarWidth = useScrollbarWidth({ targetDocument });
 
+  const renderRow: RowRenderFunction<Item> = React.useCallback(
```
Comment (layershifter) [inline packages/react-components/react-table/stories/DataGrid/Virtualization.stories.tsx:185]: nit: the confusion there is that it's not a function

I would call it `RowRenderer` and move to an upper scope

**Review by layershifter — APPROVED** [review]

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/Avatar-mount_base.html">611</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/Avatar-mount.html">605</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/Button-mount_base.html">293</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/Button-mount.html">286</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/Field-mount_base.html">1045</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/Field-mount.html">1024</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/FluentProvider-mount_base.html">647</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/FluentProvider-mount.html">650</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/FluentProviderWithTheme-mount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/FluentProviderWithTheme-mount.html">77</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">64</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">61</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">69</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/InfoButton-mount_base.html">15</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/InfoButton-mount.html">15</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/MakeStyles-mount_base.html">864</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/MakeStyles-mount.html">840</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/Persona-mount_base.html">1619</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/Persona-mount.html">1549</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/SpinButton-mount_base.html">1304</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28372/perf-test-react-components/SpinButton-mount.html">1297</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `65.162 kB`<br />`17.952 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `210.395 kB`<br />`58.717 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `36.395 kB`<br />`11.996 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.473 kB`<br />`2.196 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                                                                                 | `156.955 kB`<br />`42.834 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `44.652 kB`<br />`12.468 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>                                                                 | `132.615 kB`<br />`33.871 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr>                                                         |  `78.215 kB`<br />`19.263 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>                                                                   |  `76.891 kB`<br />`18.989 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/104c81e44da46e879cf01d709e063210350d91a5'>104c81e44da46e879cf01d709e063210350d91a5</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-table/stories/DataGrid/Virtualization.stories.tsx (hunk 1 above)
Comment (ling1726) [reply packages/react-components/react-table/stories/DataGrid/Virtualization.stories.tsx]: Makes sense, done

**Pull request merged**